### DISCUSSION TITLE
Best practise for Button click that leads to a route change

### DISCUSSION THREAD

**Discussion by VMLuca** [vercel/next.js#66123] [Help] [answered]
### Summary

Hello,

I have a button and I want to route to another page on button click.
I decided for this approach but I am wondering if this is best practise?

``` tsx
<Link href="/anotherPage">
    <Button variant="vm-primary">Select</Button>
</Link>
```

**Comment by icyJoseph** [comment, marked as answer]
Nesting button within an anchor tag is not best practice: https://caninclude.glitch.me/caninclude?child=button&parent=a it is not valid HTML

The most common way to do this, is to style the anchor tag, so that it looks like a button, or the so called, imperative way, where the button, on click, invokes the router.push or replace methods, but that is not semantic HTML, which means it is not necessarily very accessible, without more work, and it doesn't work when JavaScript is disabled

**Reply by VMLuca** [reply]
ah thanks!
